src/org/jgroups/protocols/raft/RAFT.java

```diff
@@ -858,6 +856,15 @@ else if(r instanceof DownRequest) {
                     reqtab.create(index++, raft_id, dr.f, this::majority, dr.options);
                     length+=dr.length;
                 }
+                else if (r instanceof SnapshotRequest) {
+                    SnapshotRequest sr = (SnapshotRequest) r;
+                    try {
+                        takeSnapshot(); sr.f.complete(null);
+                    } catch (Exception e) {
+                        sr.f.completeExceptionally(e);
+                        throw e;
```

Comment:
There is no need to rethrow this exception. It will run from a single thread and would discard the batch of messages.

Reply:
It's in a Throwable try catch block, it can't affect other messages of the batch, and I think the exception is a internal error of the state machine it‘s worth printing out.